Sources/Sentry/SentryTransaction.m

```diff
@@ -47,9 +47,47 @@ - (void)setMeasurementValue:(id)value forKey:(NSString *)key
     if (serializedData[@"contexts"] != nil) {
         [mutableContext addEntriesFromDictionary:serializedData[@"contexts"]];
     }
-    mutableContext[@"trace"] = [_trace serialize];
+    
+    NSMutableDictionary *traceDict = [[_trace serialize] mutableCopy];
+    
+    // We need to remove tags and data from trace, and move them to the top level
+    // https://github.com/getsentry/sentry-cocoa/issues/1311
+    NSMutableDictionary *traceTags = [NSMutableDictionary new];
+    if(traceDict[@"tags"] != nil && [traceDict[@"tags"] isKindOfClass:NSDictionary.class]) {
+        [traceTags addEntriesFromDictionary:traceDict[@"tags"]];
+        traceDict[@"tags"] = nil;
+    }
+    
+    NSMutableDictionary *traceData = [NSMutableDictionary new];
+    if(traceDict[@"data"] != nil && [traceDict[@"data"] isKindOfClass:NSDictionary.class]) {
+        [traceData addEntriesFromDictionary:traceDict[@"data"]];
+        traceDict[@"data"] = nil;
+    }
```

Comment:
`h`: Sorry, the issue has wrong information. The serialization of data is working correctly. No change is required.